src/test/java/io/crate/client/jdbc/integrationtests/CrateJDBCIntegrationTest.java

```diff
@@ -42,6 +46,26 @@ private static String clientVersion() {
         throw new RuntimeException("unable to get version of crate-client");
     }
 
+    @After
+    public void tearDown() throws Exception {
+        /**
+         * Check if no Crate client threads are left over ...
+         */
+        List<Thread> threads = new ArrayList<>();
+        for (Thread thread : Thread.getAllStackTraces().keySet()) {
+            if (thread.getName().startsWith("elasticsearch[crate-client-")) {
+                threads.add(thread);
+            }
+        }
+        if (threads.size() > 0) {
+            System.err.printf("%d thread(s) from Crate clients where left over!\n" +
```

Comment:
you'd need to track and close every single connection you open in any test method.
there are a few tests where this would add quite a bit of overhead
however, to make sure there are no leaks at all, we should probably add this assertion to the #114 fix